File: bounding_boxes.lua

require 'image'

window = image.display(image.lena()) 

-- bounding box stride and size
stridex = 50
stridey = 50
lenx = 100
leny = 100

-- threshold for bouning box success
class_thresh = 0.8

file = "243.jpg"

img = image.load(file)
local nchan, height, width = img:size(1), img:size(2), img:size(3)

for by=0, height, stridey do
	bh = leny
	if (by + bh > height) then
		bh = height - by
	end
	for bx=0, width, stridex do
		bw = lenx
		if (bx + bw > width) then
			bw = width - bx
		end
		box = image.crop(img, bx,by,bx+bw,by+bh)
		-- detect classifier here on box
		percent, strclass = math.random(), "sign"
		if (percent >= class_thresh) then
			-- add to box to the list 
			-- dict[strclass].append([percent, bx, by, bw, bh])
			-- print(percent)
		end
		image.display{image=box, win=window}
	end
end

File: bounding_boxes.py

import cv2
import random
from collections import defaultdict

# bounding box stride and size
stridex = 50
stridey = 50
lenx = 100
leny = 100

# threshold for bouning box success
class_thresh = 0.8

img = cv2.imread("243.jpg")
height, width, ch = img.shape
dict = defaultdict(list)

for by in range(0, height, stridey):
	bh = leny
	if (by + bh > width):
		bh = height - by
	for bx in range(0, width, stridex):
		bw = lenx
		if (bx + bw > width):
			bw = width - bx
		box = img[by:by+bh,bx:bx+bw]
		# detect classifier here on box
		percent, strclass = random.uniform(0,1), "sign"
		if (percent >= class_thresh):
			# add to box to the list 
			dict[strclass].append([percent, bx, by, bw, bh])
		#cv2.imshow('ROI',box)
		#cv2.waitKey(0)

# loop through the matched boxes and do something with them
# possibly prep it for 
for strclass in dict.keys():
	for box in dict[strclass]:
		cv2.rectangle(img,(box[1],box[2]),(box[1]+box[3],box[2]+box[4]), (0,255,0), 3)


cv2.imshow('ROI',img)
cv2.waitKey(0)